### PR TITLE
CFE-3874: cf-secret encrypt now encrypts for localhost if no key or host is specified.

### PR DESCRIPTION
**Pull request by larsewi**

### CODE REVIEW

**Comment by larsewi** [comment]
@cf-bottom Jenkins please :)

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7432/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7432/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7432/

**Review by vpodzime — DISMISSED** [review]
Looks good to me. Thanks!

**Comment by larsewi** [comment]
@cf-bottom Jenkins again please :)

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7433/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7433/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7433/

**Review by vpodzime — COMMENTED** [review]
Looks good to me. But please split the first commit into:
1. memory leaks fixes
2. key path fixes
3. defaulting to localhost
4. +use the new `StringIsLocalhostIP()` function you wrote

**Comment by larsewi** [comment]
@cf-bottom please check that I did not break anything in Jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7452/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7452/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7452/

**Review by olehermanse — APPROVED** [review]
LGTM

File: tests/acceptance/27_cf-secret/encrypt-no-key.cf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  files:
+    # Create plain text file
+      "$(G.testfile).plaintext"
+        create => "true",
+        content => "Hello World!";
+
+    # Delete encrypted file
+      "$(G.testfile).encrypted"
+        delete => init_delete;
```
Comment (nickanderson) [inline tests/acceptance/27_cf-secret/encrypt-no-key.cf:37]: if the encrypted file is not present doesn't this error? I think you might want to add `if => fileexists( "$(G.testfile).encrypted)`

**Review by nickanderson — APPROVED** [review]
NM on my comments, i checked and it's fine to promise a file is absent even when it doesn't exist, makes sense. lgtm.

**Review by vpodzime — APPROVED** [review]
ACK. Thanks!

**Pull request merged**